app/helpers/patient_helper.rb

```diff
@@ -2,6 +2,209 @@
+  end
+
+  # Return a string representing the birthsex of the given FHIR::Patient
+  def self.birthsex(patient)
+    url = 'us-core-birthsex'
+    code = patient&.extension&.select { |e| e.url.include?(url) }&.first&.valueCode
+    return 'Male' if code == 'M'
+    return 'Female' if code == 'F'
+    return 'Unknown' if code == 'UNK'
+
+    nil
+  end
+
+  # Helper to create an extension for preferred contact method
+  def to_preferred_contact_method_extension(preferred_contact_method)
+    preferred_contact_method.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/preferred_contact_method',
+      valueString: preferred_contact_method
+    )
+  end
+
+  # Helper to understand an extension for preferred contact method
+  def self.from_preferred_contact_method_extension(patient)
+    pcm = patient&.extension&.select { |e| e.url.include?('preferred_contact_method') }&.first&.valueString
+    pcm = nil unless ['E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].include?(pcm)
+    pcm
+  end
+
+  # Helper to create an extension for preferred contact time
+  def to_preferred_contact_time_extension(_preferred_contact_method)
+    preferred_contact_time.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/preferred_contact_time',
+      valueString: preferred_contact_time
+    )
+  end
+
+  # Helper to understand an extension for preferred contact time
+  def self.from_preferred_contact_time_extension(patient)
+    pct = patient&.extension&.select { |e| e.url.include?('preferred_contact_time') }&.first&.valueString
+    pct = nil unless %w[Morning Afternoon Evening].include?(pct)
+    pct
+  end
+
+  # Helper to create an extension for symptom onset date
+  def to_symptom_onset_date_extension(symptom_onset)
+    symptom_onset.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/symptom_onset_date',
+      valueDate: symptom_onset
+    )
+  end
+
+  # Helper to understand an extension for symptom onset date
+  def self.from_symptom_onset_date_extension(patient)
+    Date.strptime(patient&.extension&.select { |e| e.url.include?('symptom_onset_date') }&.first&.valueDate&.to_s || '', '%Y-%m-%d')
+  rescue ArgumentError
+    nil
+  end
+
+  # Helper to create an extension for last exposure date
+  def to_last_exposure_date_extension(last_exposure)
+    last_exposure.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/last_exposure_date',
+      valueDate: last_exposure
+    )
+  end
+
+  # Helper to understand an extension for last exposure date
+  def self.from_last_exposure_date_extension(patient)
+    Date.strptime(patient&.extension&.select { |e| e.url.include?('last_exposure_date') }&.first&.valueDate&.to_s || '', '%Y-%m-%d')
+  rescue ArgumentError
+    nil
+  end
+
+  # Helper to create an extension for isolation status
+  def to_isolation_extension(isolation)
+    FHIR::Extension.new(
+      url: 'http://saraalert.org/isolation',
```

Comment:
See previous comment about extension URLs. At some point we will want these to resolve, so something like `http://saraalert.org/StructureDefinition/isolation` may be a better choice (there are other options too, like making these Observations which is a more FHIR-like approach).